helper/cert/source_gen.go

```diff
@@ -393,3 +445,41 @@ func parseCert(pemValue []byte) (*x509.Certificate, error) {
 
 	return x509.ParseCertificate(block.Bytes)
 }
+
+// decodeCerts decodes a caBundle ([]byte) into a list of certs ([][]byte)
+func decodeCerts(caBundle []byte) tls.Certificate {
+	certs := tls.Certificate{}
+	remainder := caBundle
+	var next *pem.Block
+	for {
+		next, remainder = pem.Decode(remainder)
+		if next == nil {
+			break
+		}
+		if next.Type == "CERTIFICATE" {
+			certs.Certificate = append(certs.Certificate, next.Bytes)
+		}
+	}
+	return certs
+}
+
+// appendCert returns a new CA bundle:
+// [0] last CA cert from oldCABundle
+// [1] newCACert
+func appendCert(oldCABundle, newCACert []byte) ([]byte, error) {
```

Comment:
I think this name could be a bit better to capture that it's smarter than a simple append. Maybe `prependPreviousCA`, `prependLastCA`, or `includePreviousCA`, possibly with the argument order swapped? As of course the ordering is important too, because we always want to keep the CAs in age order.